SPECS-EXTENDED/linuxptp/linuxptp.spec

```diff
@@ -1,72 +1,167 @@
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
-%global gitfullver e0580929f451e685d92cd10d80b76f39e9b09a97
-%global gitver %(c=%{gitfullver}; echo ${c:0:6})
 %global _hardened_build 1
-%global testsuite_ver a7f6e1
-%global clknetsim_ver 79ffe4
+%global testsuite_ver ff37e2
+%global clknetsim_ver 9ed48d
+%global selinuxtype targeted
+%bcond_with selinux
 
 Name:		linuxptp
-Version:	2.0
-Release:	8%{?dist}
+Version:	3.1.1
+Release:	7%{?dist}
 Summary:	PTP implementation for Linux
 
 License:	GPLv2+
 URL:		http://linuxptp.sourceforge.net/
 
-#Source0:	https://downloads.sourceforge.net/%{name}/%{name}-%{version}.tgz
-Source0:	https://github.com/richardcochran/%{name}/archive/%{gitver}/%{name}-%{gitver}.tar.gz
+Source0:	https://sourceforge.net/projects/%{name}/files/v3.1/%{name}-%{version}.tgz
 Source1:	phc2sys.service
 Source2:	ptp4l.service
 Source3:	timemaster.service
 Source4:	timemaster.conf
+Source5:	ptp4l.conf
 # external test suite
-Source10:	https://github.com/mlichvar/linuxptp-testsuite/archive/%{testsuite_ver}/linuxptp-testsuite-%{testsuite_ver}.tar.gz
+Source10:	linuxptp-testsuite-%{testsuite_ver}.tar.gz
 # simulator for test suite
-Source11:	https://github.com/mlichvar/clknetsim/archive/%{clknetsim_ver}/clknetsim-%{clknetsim_ver}.tar.gz
-
-BuildRequires:	gcc gcc-c++ systemd
-BuildRequires:	net-snmp-devel
+Source11:	clknetsim-%{clknetsim_ver}.tar.gz
+# selinux policy
+Source20:	linuxptp.fc
+Source21:	linuxptp.if
+Source22:	linuxptp.te
+
+# fix handling of zero-length messages
+Patch1:		linuxptp-zerolength.patch
+# revert phc2sys options needed by the older version of test suite
+Patch2:		clknetsim-phc2sys.patch
+
+# The following patches are for HA in linuxptp
+# https://review.opendev.org/c/starlingx/integ/+/891638
+Patch3:         0001-clock-Reset-state-when-switching-port-with-same-best.patch
```

Comment:
IMHO, we could club all these patches as is in a single file and name it appropriately, e.g a place holder name like:  `enable-ha.patch`

```bash
$ cat 00*.patch > enable-ha.patch
```
What do you think ?

Reply:
That's a good idea. I will do it.